### DISCUSSION TITLE
How to make reveal_type() directive print full module path of a class?

### DISCUSSION THREAD

**Discussion by Butter934** [microsoft/pyright#1342] [Q&A]
I'm trying to use reveal_type() directive to peek the type deduced by pyright. But it looks like pyright will only print the class name without the full "path" to that class.
For example, there's a directory structure like
```
- testProject
- - A
- - - __init__.py
- - - ClassB.py
- - __init__.py
```
and the ClassB.py:
```
class ClassB:
    def __init__(self, a):
        reveal_type(self)
        self.a = a
```

When runnig:
```
> pyright .\testProject\A\ClassB.py
```
in shell.
The output looks like this:
```
E:\testProject\A\ClassB.py
  3:21 - info: Type of "self" is "ClassB"
```
Is it possible to make the output like 'testProject.A.ClassB.ClassB', as the mypy did:
```
> mypy .\testProject\A\ClassB.py --check-untyped-defs
testProject\A\ClassB.py:3: note: Revealed type is 'testProject.A.ClassB.ClassB'
```

**Comment by erictraut** [comment]
The "type printer" in Pyright doesn't include the full module path of a type. There is rarely any ambiguity, and the full module paths are verbose and hard to read, so we opted to use the short name always.

Out of curiosity, why do you want the long names? Do you have a bunch of redundant class names in your project where the origin of the type isn't obvious from context?

**Reply by Butter934** [reply]
Thanks for your explain. I'm trying to build a tool that find the files involved by a variable, while doing this by get the type deduced by type checkers, the long name expected.

**Reply by necaris** [reply]
FWIW, I'd have a use-case for this -- I'm running into situations where I have `asyncio.Future`s and `concurrent.futures.Future`s around, and can't easily tell which is which from the output.

**Reply by necaris** [reply]
See #1661
